src/natcap/invest/habitat_quality.py

```diff
@@ -188,281 +191,326 @@
-                #  * Anything other than 0 or nodata is replaced with 1
-                LOGGER.info('Preprocessing threat values for %s',
-                            aligned_threat_path)
-                threat_nodata = pygeoprocessing.get_raster_info(
-                    aligned_threat_path)['nodata'][0]
-                threat_raster = gdal.OpenEx(aligned_threat_path,
-                                            gdal.OF_RASTER | gdal.GA_Update)
-                threat_band = threat_raster.GetRasterBand(1)
-                for block_offset in pygeoprocessing.iterblocks(
-                        (aligned_threat_path, 1), offset_only=True):
-                    block = threat_band.ReadAsArray(**block_offset)
-
-                    # First check if we actually need to set anything.
-                    # No need to perform unnecessary writes!
-                    if set(numpy.unique(block)) == set([0, 1]):
-                        continue
-
-                    zero_threat = numpy.isclose(block, threat_nodata)
-                    block[zero_threat] = 0
-                    block[~numpy.isclose(block, 0)] = 1
-
-                    threat_band.WriteArray(
-                        block, yoff=block_offset['yoff'],
-                        xoff=block_offset['xoff'])
-                threat_band = None
-                threat_raster = None
+                    aligned_bound_threat_path)
+
+                # Bound threat raster values to 0 <= x <= 1
+                bound_threat_values_task = task_graph.add_task(
+                    _bound_raster_values,
+                    args=(
+                        (aligned_threat_path, 1), aligned_bound_threat_path),
+                    target_path_list=[aligned_bound_threat_path],
+                    dependent_task_list=[align_task],
+                    task_name=f'bound_threat{lulc_key}_{threat}')
+                bound_threat_values_tasks.append(bound_threat_values_task)
 
     LOGGER.info('Starting habitat_quality biophysical calculations')
-
     # Rasterize access vector, if value is null set to 1 (fully accessible),
     # else set to the value according to the ACCESS attribute
     cur_lulc_path = lulc_path_dict['_c']
     fill_value = 1.0
-    try:
-        LOGGER.info('Handling Access Shape')
-        access_raster_path = os.path.join(
-            inter_dir, 'access_layer%s.tif' % suffix)
-        # create a new raster based on the raster info of current land cover
-        pygeoprocessing.new_raster_from_base(
-            cur_lulc_path, access_raster_path, gdal.GDT_Float32,
-            [_OUT_NODATA], fill_value_list=[fill_value])
-        pygeoprocessing.rasterize(
-            args['access_vector_path'], access_raster_path, burn_values=None,
-            option_list=['ATTRIBUTE=ACCESS'])
-
-    except KeyError:
-        LOGGER.info(
-            'No Access Shape Provided, access raster filled with 1s.')
+
+    LOGGER.info('Handling Access Shape')
```

Comment:
Can you change this logger message to something more informative or drop it completely? I realize this is just highlighted from the change in indent but looks easy to "handle" now?